ts/src/pro/blofin.ts

```diff
@@ -0,0 +1,672 @@
+            this.orders = new ArrayCacheBySymbolById (limit);
+        }
+        const arg = this.safeDict (message, 'arg');
+        const channelName = this.safeString (arg, 'channel');
+        const data = this.safeList (message, 'data');
+        for (let i = 0; i < data.length; i++) {
+            const order = this.parseWsOrder (data[i]);
+            const symbol = order['symbol'];
+            const messageHash = channelName + ':' + symbol;
+            this.orders[symbol] = order;
+            client.resolve (this.orders[symbol], messageHash);
+        }
+    }
+
+    parseWsOrder (order, market: Market = undefined): Order {
+        return this.parseOrder (order, market);
+    }
+
+    async watchPositions (symbols: Strings = undefined, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Position[]> {
+        /**
+         * @method
+         * @name blofin#watchPositions
+         * @see https://docs.blofin.com/index.html#ws-positions-channel
+         * @description watch all open positions
+         * @param {string[]|undefined} symbols list of unified market symbols
+         * @param {object} params extra parameters specific to the exchange API endpoint
+         * @returns {object[]} a list of [position structure]{@link https://docs.ccxt.com/en/latest/manual.html#position-structure}
+         */
+        await this.authenticate ();
+        await this.loadMarkets ();
+        const newPositions = await this.watchMultipleWrapper (false, 'positions', 'watchPositions', symbols, params);
+        if (this.newUpdates) {
+            return newPositions;
+        }
+        return this.filterBySymbolsSinceLimit (this.positions, symbols, since, limit);
+    }
+
+    handlePositions (client: Client, message) {
+        //
+        //     {
+        //         arg: { channel: 'positions' },
+        //         data: [
+        //           <same object as shown in REST example>
+        //         ]
+        //     }
+        //
+        if (this.positions === undefined) {
+            this.positions = new ArrayCacheBySymbolBySide ();
+        }
+        const cache = this.positions;
+        const arg = this.safeDict (message, 'arg');
+        const channelName = this.safeString (arg, 'channel');
+        const data = this.safeList (message, 'data');
+        const newPositions = [];
+        for (let i = 0; i < data.length; i++) {
+            const position = this.parseWsPosition (data[i]);
+            newPositions.push (position);
+            cache.append (position);
+            const messageHash = channelName + ':' + position['symbol'];
+            client.resolve (position, messageHash);
+        }
+    }
+
+    parseWsPosition (position, market: Market = undefined): Position {
+        return this.parsePosition (position, market);
+    }
+
+    async watchMultipleWrapper (isPublic: boolean, channelName: string, callerMethodName: string, symbolsArray: any[] = undefined, params = {}) {
```

Comment:
@ttodua next time please let's avoid methods like this one, this is very hard to maintain and track and very error prone (in fact `watchorders` is not even working...)